src/main/java/com/syncapse/jenkinsci/plugins/awscloudformationwrapper/CloudFormation.java

```diff
@@ -248,273 +247,259 @@
-        Hudson hudson = Hudson.getInstance(); 
+        AWSCredentialsProvider provider = new AWSStaticCredentialsProvider(credentials);
+        AmazonCloudFormationAsyncClientBuilder builder = AmazonCloudFormationAsyncClientBuilder.standard();
+        builder.withCredentials(provider);
+
+        Hudson hudson = Hudson.getInstance();
         ProxyConfiguration proxyConfig = hudson != null ? hudson.proxy : null;
         if (proxyConfig != null && proxyConfig.name != null) {
            ClientConfiguration config = new ClientConfiguration();
~           config.setProxyHost(proxyConfig.name);
~           config.setProxyPort(proxyConfig.port);
            config.setProxyUsername(proxyConfig.getUserName());
            config.setProxyPassword(proxyConfig.getPassword());
            config.setPreemptiveBasicProxyAuth(true);
-           AWSCredentialsProvider provider = new BasicAWSCredentialsProvider(credentials);
-           AmazonCloudFormation amazonClient = new AmazonCloudFormationAsyncClient(
-                   provider, config);
-   
-           amazonClient.setEndpoint(awsRegion.endPoint);
-           return amazonClient;
-        } else {
-          AmazonCloudFormation amazonClient = new AmazonCloudFormationAsyncClient(
-                  credentials);
-          amazonClient.setEndpoint(awsRegion.endPoint);
-          return amazonClient;
+           builder.withClientConfiguration(config);
         }
+
+        builder.setRegion(this.awsRegion.awsRegion);
+        return builder.build();
     }
 
     private boolean waitForStackToBeDeleted() {
~        int retries = 1;
~        Stack stack = null;
~        while (true) {
~            try {
~
~              stack = getStack(amazonClient.describeStacks());
~
~              if (stack == null) {
~                  return true;
~              }
~
~              StackStatus stackStatus = getStackStatus(stack.getStackStatus());
~
~              if (StackStatus.DELETE_COMPLETE == stackStatus) {
~                  return true;
~              }
~
~              if (StackStatus.DELETE_FAILED == stackStatus) {
~                  return false;
               }
 
               logger.println("Stack status " + stackStatus + ".");
-              
+
             } catch (AmazonServiceException ase) {
                 if (!RetryUtils.isThrottlingException(ase)) {
                     throw ase;
~                }
~            }
~            sleep(stack, retries);
~            retries++;
~        }
~
~    }
~
~    private List<Parameter> parameters(Map<String, String> parameters) {
~
~        if (parameters == null || parameters.values().size() == 0) {
~            return null;
~        }
~
~        List<Parameter> result = Lists.newArrayList();
~        for (Map.Entry<String, String> entry : parameters.entrySet()) {
~            Parameter parameter = new Parameter();
~            parameter.setParameterKey(entry.getKey());
~            parameter.setParameterValue(entry.getValue());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.